### DISCUSSION TITLE
Arctic Code Vault lost badge when I deleted my account & changed my username

### DISCUSSION THREAD

**Discussion by barryanders** [community/community#20828] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi, I lost the arctic code vault badge when I changed my username. More specifically, I actually deleted my account after moving my repositories to a new account. I did it somewhat hastily and I didn't consider the fact that all of my past commits to other's repositories (not many but some) would no longer be associated with my account. I believe the commit(s) that got me the badge was in contribution to @mathiasbynens [dotfiles](https://github.com/mathiasbynens/dotfiles). The deleted account was `BarryMode`. Is there any way to recover the badge?

Edit: here is the pull request I remember for this: https://github.com/mathiasbynens/dotfiles/pull/886

**Comment by rodukov** [comment, marked as answer]
I think this achievement was given to a specific usernames. **Not to their accounts.** I see two solutions:
1) Return the username to its original format;
2) Write to support.

**Reply by tobiasvl** [reply]
I doubt it was given to specific usernames, it's probably because OP deleted their account rather than rename it. And that's not fixable.

**Reply by rodukov** [reply]
> I doubt it was given to specific usernames, it's probably because OP deleted their account rather than rename it. And that's not fixable.

Yes you are right.

**Reply by barryanders** [reply]
> I think this achievement was given to a specific usernames. **Not to their accounts.** I see two solutions:
> 
> 1) Return the username to its original format;
> 
> 2) Write to support.

Hey, thanks for your help. Support looks like they're going to work it out!

**Comment by tobiasvl** [comment]
You didn't change your username, you deleted your account. [GitHub can't restore your deleted account.](https://docs.github.com/en/enterprise-server@3.2/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/deleting-your-personal-account)

Git commits are tied to e-mail addresses, not GitHub usernames. Is the e-mail address you used for the commit in the PR associated with your new username? If not, you should add it to your new account as an alternate e-mail address, that should tie the commit to you and _perhaps_ grant you the achievement badge again: https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/adding-an-email-address-to-your-github-account

However, the badge was granted at a specific time, and might not be earnable after the fact. You could contact [GitHub support](https://support.github.com/) and explain the situation though.

**Reply by radar-blip** [reply]
I concur.

**Comment by jfullstackdev** [comment]
the badge is linked to the email when you achieved that, as long as it is not yet taken or in your case deleted, try adding it even unverified, the problem is when you don't know that email linked to that badge

but I'm not very sure when you delete your account whether it will free the email

**Reply by barryanders** [reply]
Unfortunately, I can't add the email because I think I used the noreply email associated with the account I deleted, lol. Looks like I put myself in an impossible situation here.

Edit: I was able to add the original GPG key back to my account. The email isn't verified, but at least that's some kind of progress. I found the GPG key in a backup.

**Reply by barryanders** [reply]
As it turns out, I was wrong. Support found that it was a different email of mine that I still have access to... they made it sound like if I can verify the email that they will fix it! So I verified it.
